src/ASM/ASMbase.h

```diff
@@ -165,6 +165,8 @@ class ASMbase
   virtual unsigned char getNoFields(int b = 0) const { return b > 1 ? 0 : nf; }
   //! \brief Returns the number of Lagrange multipliers per node.
   unsigned char getNoLagPerNode() const { return nLag; }
+  //! \brief Returns the number of bases.
+  virtual size_t getNoBasis() const { return 1; }
```

Comment:
we do not need another one! https://github.com/OPM/IFEM/blob/master/src/ASM/ASMbase.h#L235